### ISSUE TITLE
Add support for letters and numbers in list

### ISSUE THREAD

**Issue by D-Marc1** [michelf/php-markdown#292]
I'd love to see support for something for like this.

1.
2.
2a.
3.

A.
B.
C.

**Comment by jerry1970** [comment]
That would be possible if we could add a class to a numbered list. Is it?

Using the `list-style-type` style you can style the list. Examples:
https://css-tricks.com/numbering-in-style/

So the question is how we could add a class to a list.

When you do this:
```
{.class-name}
1. test
1. test
```
you get a `<p class="class-name">` and no list. When you write `1. {.test} Test`, the class name is not parsed but taken literally.

I have made an extension that adds "fenced blocks". Not fenced code blocks but fenced blocks. Only thing that does is wrap a `<div>` around the fenced block and you can add classes to that. Something like this:

```
...{.someclass}
1. test
1. test
1. test
...
```
The `...` marks the fenced block. Unlike fenced code blocks, the contents of the fenced blocks are parsed, so the result is this:
```
<div class="someclass">
<ol>
<li>test</li>
<li>test</li>
<li>test</li>
</ol>
</div>
```
I can target CSS on `.someclass ol` now.

**Comment by michelf** [comment]
@D-Marc1 Sorry for not replying sooner... Adding other list types is not necessarily out of question, but your "2a." item confuses me. What kind of HTML would that produce?

@jerry1970 I think the goal here is to have a convenient way to have custom list variants. Inventing a syntax so you can wrap the list in a `div` with attributes so it can achieve the desired result in combination with a proper external style sheet is a bit convoluted, but if it's a `div` you want there's already a syntax for this:

~~~
<div class="someclass" markdown="1">
1. test
1. test
</div>
~~~

**Comment by lacymorrow** [comment]
I have a need for this edge case.

I'm creating legal pages which have lists similar to below and don't have a clear method of marking this up without resorting to HTML
```
1. Intro
	1.1 ...
	1.2 ...
		1.2.1 ...
		1.2.2 ...
		1.2.3 ...
	1.3 ...
		a) ...
		b) ...
2. Overview
	a) ...
	b) ...
```

**Comment by michelf** [comment]
I'm thinking about it, and while I'm not sure parsing all this should be active by default (worrying about false positives), I think it could be enabled with a configuration variable.

A challenge though is that some of these will require CSS in the output to work, or predefined class names and a matching stylesheet. Again, something that probably should be configurable in some way.

**Comment by lacymorrow** [comment]
I agree it's a rare case, and looking further it seems this isn't supported by regular Markdown. It may be outside the scope of this project or Markdown in general, since Markdown manipulates HTML, HTML would require a CSS class to achieve this, and that sounds dangerously like "concerning itself with style."
